src/client/src/app/site/[site]/cancel-day/cancel-day-form.test.tsx

```diff
@@ -0,0 +1,104 @@
+import { screen } from '@testing-library/react';
+import CancelDayForm from './cancel-day-form';
+import render from '@testing/render';
+import { useRouter } from 'next/navigation';
+import { parseToUkDatetime } from '@services/timeService';
+import * as appointmentsService from '@services/appointmentsService';
+
+jest.mock('next/navigation', () => ({
+  useRouter: jest.fn(),
+}));
+
+jest.mock('@services/timeService', () => ({
+  parseToUkDatetime: jest.fn(),
+}));
+
+jest.mock('@components/session-summary-table', () => ({
+  SessionSummaryTable: ({ tableCaption }: { tableCaption: string }) => (
+    <div data-testid="session-summary">{tableCaption}</div>
+  ),
+}));
+
+jest.mock('@services/appointmentsService');
+const mockCancelDay = jest.spyOn(appointmentsService, 'cancelDay');
+
+const mockReplace = jest.fn();
+
+const defaultProps = {
+  date: '2025-01-01',
+  siteId: 'site-123',
+  daySummary: {
+    date: '2025-01-01',
+    maximumCapacity: 10,
+    remainingCapacity: 7,
+    bookedAppointments: 3,
+    orphanedAppointments: 0,
+    cancelledAppointments: 0,
+    sessions: [],
+  },
+  clinicalServices: [],
+};
+
+describe('CancelDayForm', () => {
+  beforeEach(() => {
+    (useRouter as jest.Mock).mockReturnValue({ replace: mockReplace });
+    (parseToUkDatetime as jest.Mock).mockReturnValue({
+      format: () => 'Wednesday 1 January',
+    });
+    jest.clearAllMocks();
+  });
+
+  it('renders the session summary and inset text', () => {
+    render(<CancelDayForm {...defaultProps} />);
+    expect(screen.getByTestId('session-summary')).toHaveTextContent(
+      'Sessions for Wednesday 1 January',
+    );
+    expect(
+      screen.getByText(/3 booked appointments will be cancelled/i),
+    ).toBeInTheDocument();
+  });
+
+  it('initially disables Continue until a choice is made', () => {
+    render(<CancelDayForm {...defaultProps} />);
+    expect(screen.getByRole('button', { name: /continue/i })).toBeDisabled();
+  });
+
+  it('allows selecting No and navigates back on Continue', async () => {
+    const { user } = render(<CancelDayForm {...defaultProps} />);
+
+    await user.click(screen.getByLabelText(/no, i don't want/i));
+    const continueBtn = screen.getByRole('button', { name: /continue/i });
+
+    expect(continueBtn).toBeEnabled();
+
+    await user.click(continueBtn);
+    expect(mockReplace).toHaveBeenCalledWith(
+      `/site/site-123/view-availability/week?date=2025-01-01`,
+    );
+  });
+
+  it('allows selecting Yes and shows confirmation step', async () => {
+    const { user } = render(<CancelDayForm {...defaultProps} />);
+
+    await user.click(screen.getByLabelText(/yes, i want to cancel/i));
+    await user.click(screen.getByRole('button', { name: /continue/i }));
```

Comment:
Bit of a minor one, but there's a performance hit to using regex so I'd try to only use it where necessary. We have quite a lot of tests now so probably worth noting. Most (all?) of the tests in this file still pass without needing the pattern matching - 
<img width="769" height="178" alt="image" src="https://github.com/user-attachments/assets/4ae80175-f5c1-4e49-829e-275adf177b0e" />